### ISSUE TITLE
Confirm Current Version Matrix is Correct

### ISSUE THREAD

**Issue by mpchadwick** [elastic/elasticsearch-php#1031]
### Summary of problem or feature request

The version matrix was last updated 6 months ago and says that for Elasticsearch >= 7.4 the 7.4 branch should be used.

https://github.com/elastic/elasticsearch-php/blame/6c9c0a765f8b31c1d9f1a5af4de2f53fd7298e38/README.md#L55

7.4 is now a "stale branch" https://github.com/elastic/elasticsearch-php/branches and there are tags for the 7.6 and 7.7.

Is the 7.4 branch still best to use for any Elasticsearch version >= 7.4? I assume this needs to be updated?

---

Heres' a specific example of where this is causing some confusion... Magento added support for Elasticsearch 7.2 and 7.5 in v2.3.5 of their product (https://github.com/magento/devdocs/pull/7133/files#diff-d9af9c4b7eef4dec2fa8fd42dad9a6aeR163). However they're currently constraining `elasticsearch/elasticsearch` to 7.6 in their codebase: https://github.com/magento/magento2/blob/dcc962efc5ad12717c6eac0f10cf822780dd3f4a/app/code/Magento/Elasticsearch7/composer.json#L11. The version matrix isn't currently making it clear if Magento's code base contains the wrong version in their `composer.json` or if it should be expected to work correctly.

### System details

- Operating System
- PHP Version
- ES-PHP client version
- Elasticsearch version

**Comment by ezimuel** [comment]
@mpchadwick sorry for my late reply :disappointed:  You right, I updated the version matrix [here](https://github.com/elastic/elasticsearch-php#version-matrix). Basically, we are releasing a new version of `elasticsearch-php`  each time we have a new major or minor release of Elasticsearch. For instance, now Elasticsearch is `7.8` and we released [elasticsearch-php 7.8](https://github.com/elastic/elasticsearch-php/releases/tag/v7.8.0). 
We use the same version number of Elasticsearch to be aligned with the Elastic stack. In this way, we can update all the APIs of the upcoming version of Elasticsearch without the need to have a support matrix. If there are patch releases of `Elasticsearch 7.8.x` we don't usually release a new version of the client since the compatibility is guarantee.
Long story short, if you are using `Elasticsearch X.Y` you should always use `elasticsearch-php X.Y`.
